vue-project/src/components/AddUser.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <form>
+    <custom-label labelText="Username" class="label" />
+    <custom-input
+      placeHolderText="Enter username..."
+      @get-input="getUsername"
+      inputType="text"
+      @writing="removeAlerts"
+    />
+    <div id="divMessage">
+      <fail-component v-if="failedUsername" :content="usernameErrorMessage" />
+    </div>
+    <custom-label labelText="Email" class="label" />
+    <custom-input
+      placeHolderText="Enter email..."
+      @get-input="getEmail"
+      inputType="email"
+      @writing="removeAlerts"
+    />
+    <div id="divMessage">
+      <fail-component v-if="failedEmail" :content="emailErrorMessage" />
+    </div>
+    <custom-label labelText="Password" class="label" />
+    <custom-input
+      placeHolderText="Enter password..."
+      @get-input="getPassword"
+      inputType="password"
+      @writing="removeAlerts"
+    />
+    <div id="divMessage">
+      <fail-component v-if="failedPassword" :content="passwordErrorMessage" />
+    </div>
+    <custom-button btnValue="Add" class="addButton" @click="addUser" />
+    <div id="divMessage">
+      <fail-component v-if="failedSubmit" :content="submitErrorMessage" />
+      <success-component
+        v-if="succeededSubmit"
+        :content="submitSuccessMessage"
+      />
+    </div>
+  </form>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: "AddUser",
+  props: {
+    callHideAlerts: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      username: "",
+      email: "",
+      password: "",
+      failedUsername: false,
+      failedEmail: false,
+      failedPassword: false,
+      failedSubmit: false,
+      succeededSubmit: false,
+      usernameErrorMessage: "",
+      emailErrorMessage: "",
+      passwordErrorMessage: "",
+      submitErrorMessage: "",
+      submitSuccessMessage: "",
+      emailRegex: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
```

Comment:
Add this to a helper class so it can be reused across multiple views in the future